src/openzaak/components/documenten/tests/import/test_import_documents.py

```diff
@@ -0,0 +1,817 @@
+# SPDX-License-Identifier: EUPL-1.2
+# Copyright (C) 2019 - 2022 Dimpact
+import csv
+import tempfile
+from io import StringIO
+from pathlib import Path
+from unittest.mock import patch
+from uuid import uuid4
+
+from django.db import IntegrityError, OperationalError
+from django.test import TestCase, override_settings
+
+import requests_mock
+from zgw_consumers.constants import APITypes
+from zgw_consumers.models import Service
+
+from openzaak.components.documenten.import_utils import DocumentRow
+from openzaak.components.documenten.models import EnkelvoudigInformatieObject
+from openzaak.components.documenten.tasks import import_documents
+from openzaak.components.documenten.tests.factories import (
+    DocumentRowFactory,
+    EnkelvoudigInformatieObjectFactory,
+)
+from openzaak.components.documenten.tests.utils import (
+    get_catalogus_response,
+    get_informatieobjecttype_response,
+)
+from openzaak.components.zaken.tests.factories import ZaakFactory
+from openzaak.import_data.models import (
+    ImportRowResultChoices,
+    ImportStatusChoices,
+    ImportTypeChoices,
+)
+from openzaak.import_data.tests.factories import ImportFactory
+from openzaak.import_data.tests.utils import ImportTestFileMixin
+from openzaak.tests.utils.mocks import MockSchemasMixin
+from openzaak.utils.fields import get_default_path
+
+
+@override_settings(
+    ALLOWED_HOSTS=["testserver"],
+    IMPORT_DOCUMENTEN_BASE_DIR=tempfile.mkdtemp(),
+    IMPORT_DOCUMENTEN_BATCH_SIZE=2,
+)
+class ImportDocumentTestCase(ImportTestFileMixin, MockSchemasMixin, TestCase):
+    mocker_attr = "requests_mock"
+
+    @classmethod
+    def setUpTestData(cls):
+        super().setUpTestData()
+
+        cls.catalogus = "https://externe.catalogus.nl/api/v1/catalogussen/1c8e36be-338c-4c07-ac5e-1adf55bec04a"
+        cls.informatieobjecttype = (
+            "https://externe.catalogus.nl/api/v1/informatieobjecttypen/"
+            "b71f72ef-198d-44d8-af64-ae1932df830a"
+        )
+
+        Service.objects.create(
+            api_root="https://externe.catalogus.nl/api/v1/", api_type=APITypes.ztc
+        )
+
+    def setUp(self):
+        self.requests_mock = requests_mock.Mocker()
+        self.requests_mock.start()
+
+        self.requests_mock.get(
+            self.informatieobjecttype,
+            json=get_informatieobjecttype_response(
+                self.catalogus, self.informatieobjecttype
+            ),
+        )
+        self.requests_mock.get(
+            self.catalogus,
+            json=get_catalogus_response(self.catalogus, self.informatieobjecttype),
+        )
+
+        self.addCleanup(self.requests_mock.stop)
+
+        super().setUp()
+
+    def test_simple_import(self):
+        zaken = {1: ZaakFactory(), 2: ZaakFactory()}
+
+        import_data = {
```

Comment:
Could you please create a csv file with test data and save it into test folder, so we would have an example of what the import endpoind expects

Reply:
I actually removed the (manually added) test files as it can take up quite some time to modify tests (e.g keep the values in the import file and the data from the DB in sync, updated header names, reorder headers/columns) whenever a field is added and or renamed in the import file.

Note that the files are still generated though through the `DocumentRowFactory` ([link](https://github.com/open-zaak/open-zaak/pull/1654/files#diff-f1bd1fc46084251b322c10fdc624ca950523e3df317b0aa157c0fdde9fdb4659R218)).

Reply:
See https://github.com/open-zaak/open-zaak/pull/1654/commits/3fff740c8c55fb48a7bc5135c0a6c9747a4366ce